### ISSUE TITLE
Unhelpful error message when running DDL query.

### ISSUE THREAD

**Issue by sdbip** [vapor/postgres-nio#294]
### Describe the bug

I suppose I made a mistake, but I'm not told what it was. Instead the application crashes with a fatal error.
I can run this query successfully:

```swift
_ = try await connection.query("SELECT 1", logger: logger)
```

… but when I try to create a schema in my database, I get the fatal error:

```swift
_ = try await connection.query("CREATE TABLE TestTable(value INT)", logger: logger)
```
> Thread 3: Assertion failed: PostgresConnection deinitialized before being closed.

I don't know how to correct my mistake. Is my configuration wrong? Is there something wrong in my SQL? Did I skip an important step in setting up the connection? Or is it something else that I don't even know to think of?

### To Reproduce

Run this test case:

```swift
let config = PostgresConnection.Configuration(
    connection: .init(
        host: "localhost",
        port: 5432
    ),
    authentication: .init(
        username: "johan",
        database: "test",
        password: ""
    ),
    tls: .disable
)

class PostgresTests: XCTestCase {
    
    func test_createDatabase() async throws {
        let logger = Logger(label: "postgres-logger")
        let eventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: 1)
        let connection = try await PostgresConnection.connect(on: eventLoopGroup.next(), configuration: config, id: 1, logger: logger)

        // This succeeds
        _ = try await connection.query("SELECT 1", logger: logger)

        // This crashes with "Thread 3: Assertion failed: PostgresConnection deinitialized before being closed."
        _ = try await connection.query("CREATE TABLE TestTable(value INT)", logger: logger)

        try await connection.close()
    }
}
```

### Expected behavior

* I expect a table to be created in the Postgres database, but I suppose I made a mistake that I don't currently understand.
* The connection should remain open, and not be deinitialized, even if the table couldn't be created for whatever reason.
* The error should be thrown back to the caller and provide feedback that helps me fix my mistake.

### Environment

* OS Version: macOS Monterey 12.4 (21F79)
* Xcode Version: 13.4 (13F17a)
* Postgres Desktop Version: 2.5.6 (136)
* PostgreSQL Database Version: 12
* PostgresNIO Version: 1.10.0
* Vapor: No

### Additional context

I'm not using Vapor at this time. This is just using a regular XCTest test case to connect to PostgreSQL for its own sake.

**Comment by sdbip** [comment]
I must have used the `tearDown()` override a million times, yet I didn't know about `addTearDownBlock`. Something new learned today, thank you! Is this a new feature in Xcode 13.4.1?

The problem I wanted to point out was that when `deinit` traps, it hides what goes wrong with my `CREATE TABLE` query. Is there a reason why it doesn't just close the connection if it's active?

**Comment by fabianfett** [comment]
> I must have used the tearDown() override a million times, yet I didn't know about addTearDownBlock. Something new learned today, thank you! Is this a new feature in Xcode 13.4.1?

This feature is available since Swift 5.5, however it was unusable... It really only started to work with Xcode 13.4.1

> The problem I wanted to point out was that when `deinit` traps, it hides what goes wrong with my `CREATE TABLE` query. Is there a reason why it doesn't just close the connection if it's active?

Closing a connection takes time, since it needs to send out connection close packets on tcp. If we were to do this in a `deinit` we would need to block the thread the deinit happens on. I know that explicit lifecycle management feels like a big pain at first, but it makes your resources very explicit.

If you are interested, you can write yourself a simple with function:

```
extension PostgresConnection {
  func with<Result>(config: PostgresConnection.Configuration, _ closure: (PostgresConnection) async throws -> Result) async throws -> Result {
    // 1. create connection

    // 2. invoke closure

    // 3. close connection even if closure has thrown
  }
}
```
